### PR TITLE
PWN-3685 - add UiKit snackbar component

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-3685

## Description of Work

UiKit snackbar component

## Screenshots (Optional)
Screen Shot 2022-06-09 at 2 37 21 pm

Screen Shot 2022-06-09 at 2 37 40 pm

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/common/ui/widget/CustomSnackBar.kt [hunk 1]
```diff
@@ -1,30 +1,29 @@
...
 import com.google.android.material.snackbar.ContentViewCallback
 import org.p2p.wallet.databinding.WidgetSnackbarBinding
+import org.p2p.wallet.utils.NoOp
 import org.p2p.wallet.utils.getString
+import org.p2p.wallet.utils.viewbinding.inflateViewBinding
 
 private const val ANIMATION_DURATION = 500L
 
-class SnackBar @JvmOverloads constructor(
+class CustomSnackBar @JvmOverloads constructor(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/common/ui/widget/CustomSnackBar.kt:20]: renamed due to conflicts with the androidx.Snackbar

File: ui-kit/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <style name="UiKit.Native.Snackbar" parent="Widget.MaterialComponents.Snackbar">
```
Comment (gslevinkov) [inline ui-kit/src/main/res/values/styles.xml:3]: hows the idea to name UiKit styles like this?

Comment (kuznetsov) [reply ui-kit/src/main/res/values/styles.xml:3]: Looks nice!

File: ui-kit/src/main/kotlin/org/p2p/uikit/natives/UiKitSnackbar.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        this,
+        text = snackbarText,
+        buttonText = snackbarActionButtonText,
+        buttonAction = snackbarActionButtonListener,
+        duration = Snackbar.LENGTH_SHORT
+    )
+        .show()
+}
+
+fun View.showSnackbarLong(snackbarText: CharSequence) {
```
Comment (davran1001) [inline ui-kit/src/main/kotlin/org/p2p/uikit/natives/UiKitSnackbar.kt:36]: I think duration can be optionally, and there is no reasons to duplicate methods just for this ))

**Review by davran1001 — APPROVED** [review]

**Pull request merged**